lib/sync/ephy-sync-service.c

```diff
@@ -606,15 +606,62 @@ ephy_sync_service_certificate_is_valid (EphySyncService *self,
 }
 
 static void
+forget_secrets_cb (SecretService *service,
+                   GAsyncResult  *result,
+                   gpointer       user_data)
+{
+  GError *error = NULL;
+
+  secret_service_clear_finish (service, result, &error);
+  if (error) {
+    g_warning ("Failed to clear sync secrets: %s", error->message);
+    g_error_free (error);
+  } else {
+    LOG ("Successfully cleared sync secrets");
+  }
+}
+
+static void
+ephy_sync_service_forget_secrets (EphySyncService *self)
+{
+  GHashTable *attributes;
+  char *user;
+
+  g_assert (EPHY_IS_SYNC_SERVICE (self));
+  g_assert (self->secrets);
+
+  user = ephy_sync_utils_get_sync_user ();
+  g_assert (user);
```

Comment:
Is it safe to assert here? We should not crash on invalid input; for example, if the user has messed up his GSettings somehow. Maybe this should be `g_warning` instead?